JASP-Desktop/engine/enginerepresentation.h

```diff
@@ -137,6 +153,9 @@ public slots:
 	std::string		_lastCompColName	= "???";
 
 
+	bool _runsAnalysis;
+	bool _runsUtility;
+	bool _runsRCmd;
```

Comment:
The state of the engine is quite difficult to catch, between the _engineState, _pauseRequested, _stopRequested, _slaveCrashed, and now these 3 new booleans.
Can't the engineState be extended?

Reply:
No the state of the engine is always stored in `_engineState`, that hasn't changed. But it describes what `JASP-Engine` is actually doing at the moment.

`_pauseRequested` and `_stopRequested` can not be *in* the state because there the engine is actually in another state (possibly anyway). Next process-cycle these are checked and then once that is done it sends a `pauseRequested` or `stopRequested` msg and sets *that* in the `_engineState`. Because then the engine is doing that and once the reply comes the state is set to `paused` or `stopped`. This is a bit complicated but the result of not having a msgbuffer. 

These 3 new booleans are not the state at all by the way, they simply define what this engine is used for. You remember the loop where `engine #0` is used for some things and the other engines for other things? I have simply made explicit what they are allowed to do. Because this way you can add or remove engines as desired. This will also be used in running engines more dynamically (for instance per module).

I agree that the "state of the engine" is difficult to catch but, it needs to do many things... A good solution could be to implement a msgbuffer over http or something. Then a lot of things here can be simplified. But that would be a big big change.And then those `runsAnalysis` or `runsRCmd` booleans are still useful.

Reply:
These boolean should at least have a default value.
I never see in the code setRunsCmd(true), or setRunsUtility(true)...

Reply:
Yes they should! I forgot to add that...

Reply:
Oof, I dropped a bunch of balls there